app/models/namespace.rb

```diff
@@ -225,4 +225,32 @@ def validate_nesting_level
   def restore_routes
     Route.restore(Route.only_deleted.find_by(source_id: id).id, recursive: true)
   end
+
+  def subtract_from_metadata_summary_count(namespaces, metadata, update_by_one)
```

Comment:
Could you please add a function comment block to `subtract_from_metadata_summary_count` with an explanation of the different fields and how `update_by_one` works/is used. Your explanation to @ksierks 's question makes sense, but I think we should add that info to the function.

Reply:
Added in 422c2be245bdf39d1f2475382065ee11a0c855e5. It's quite long since I wanted to ensure whenever we come back to it in the future, it's clear what the thought process was. Let me know if it's too much.